src/mpid/ch4/netmod/ofi/ofi_am_impl.h

```diff
@@ -373,6 +376,7 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_OFI_deferred_am_isend_enqueue(int rank, MPIR_
                                                          MPL_MEM_OTHER);
     MPIR_ERR_CHKANDJUMP(!deferred_req, mpi_errno, MPI_ERR_OTHER, "**nomem");
 
+    deferred_req->op = op;
```

Comment:
I know this is part of the old code, but since we are extending it -- 

I find so much passing and copying parameters back and forth as well ass keeping track of new struct very difficult to keep track... why don't we simply create `sreq` object, which we will create anyway, set the information inside the `sreq`, and simply enqueue the `sreq`?